ghastcoiler/minions/rank_2.py

```diff
@@ -0,0 +1,201 @@
+                         base_attack=3,
+                         base_defense=3,
+                         types=[MinionType.Demon],
+                         base_taunt=True,
+                         base_deathrattle=ImprisonerDeathrattle(),
+                         **kwargs)
+
+
+class KaboomBot(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Kaboom Bot",
+                         rank=2,
+                         base_attack=2,
+                         base_defense=2,
+                         types=[MinionType.Mech],
+                         base_deathrattle=KaboomBotDeathrattle(),
+                         **kwargs)
+
+
+class KindlyGrandmother(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Kindly Grandmother",
+                         rank=2,
+                         base_attack=1,
+                         base_defense=1,
+                         types=[MinionType.Beast],
+                         base_deathrattle=KindlyGrandmotherDeathrattle(),
+                         **kwargs)
+
+
+class MetaltoothLeaper(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Metaltooth Leaper",
+                         rank=2,
+                         base_attack=3,
+                         base_defense=3,
+                         types=[MinionType.Mech],
+                         # TODO : add battlecry
+                         **kwargs)
+
+
+class MurlocWarleader(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Murloc Warleader",
+                         rank=2,
+                         base_attack=3,
+                         base_defense=3,
+                         types=[MinionType.Murloc],
+                         **kwargs)
+
+    def gives_attack_defense_bonus(self, other_minion):
+        return 2, 0
+
+
+class NathrezimOverseer(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Nathrezim Overseer",
+                         rank=2,
+                         base_attack=2,
+                         base_defense=3,
+                         types=[MinionType.Demon],
+                         **kwargs)
+
+
+class OldMurkEye(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="OldMurkEye",
+                         rank=2,
+                         base_attack=2,
+                         base_defense=4,
+                         types=[MinionType.Murloc],
+                         **kwargs)
+
+    def additional_attack(self, own_board, opposing_board):
+        # TODO: +1 attack for each other murloc on the battlefield
+        pass
+
+
+class PogoHopper(Minion):
+    def __init__(self, **kwargs):
+        super().__init__(name="Pogo-Hopper",
+                         rank=2,
+                         base_attack=1,
+                         base_defense=1,
+                         types=[MinionType.Mech],
+                         **kwargs)
+
+        # TODO add battlecry
```

Comment:
Battlecries are not relevant (yet) because they only trigger during the shopping phase